[backport/2.11] Debugger readline support (#8999)
tarantool/tarantool: changelogs/unreleased/gh-7593-console-debugger.md

## feature/debugger

> * Introduced readline support to tarantool debugger (gh-7738).

Reviewer: ```suggestion
* Introduced `readline` support for the Tarantool debugger (gh-7738).
```

Author: This is actually a backport, so according to [this](https://github.com/tarantool/tarantool/pull/8880#discussion_r1266507421) discussion, no changes to changelog should be made. But it turns out, the original changelog slipped in unchecked, because I've modified it after it was approved and forgot to rerequest review.

Author: @andreyaksenov, please confirm that it is ok to leave as is.
